compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -262,7 +262,7 @@ static LogicalResult setContractConfig(func::FuncOp entryPoint,
                                    partitionedLoops.end());
         for (auto loopID : llvm::seq<unsigned>(0, numParallelLoops)) {
           if (!partitionedLoopsSet.count(loopID)) {
-            workgroupTileSizes[loopID] = 0;
+            workgroupTileSizes[loopID] = 1;
```

Comment:
I think the root issue is the kNumMaxParallelDims == 3 and the problematic case has 5 dimensions and four are a parallel loop. So the check whether the loop ID is not in the partitionable loops set is not very accurate because the batch dimension is also parallel but not in the partitionable loops set. OTOH, setting it to 1 is also not correct because then the reduction dimension would get tile size 1 instead of 0.

I think a better solution would be to reshape parallel dimensions into one to make the number of parallel dimensions always less than or equal to kNumMaxParallelDims. It would be nice to have a pass for that.

Reply:
I dont think we should be adding reshapes here... let me take a look at this PR.

Reply:
Ok, if you want to get all the partitionable loops of this operation at line 257 (above) you should just be using 

```
.getPartitionableLoops()
```

instead of

```
.getPartitionableLoops(kNumMaxParallelDims)
```

That will give you all the partitionable loops. Then you can set all the outer dimensions to be tile size of 1.

Reply:
Yes we shouldn't tile the dimensions non partitionable. This change should be reverted.

Reply:
> @ThomasRaoux just to clarify, the change you want reverted is setting the `workgroupTileSizes` to 1 along with the updates to tests (but not the `getPartitionableLoops` change)?

correct